### DISCUSSION TITLE
1.17 require-resource-pack fix bypass

### DISCUSSION THREAD

**Discussion by Tizooo** [PurpurMC/Purpur#516] [__Dismissed__] [answered]
so in 1.17 in the server.properties there is a option called `require-resource-pack` it makes it so if players dont accept the server ressourcepack they will get kickt from the server and if they accept it they can stay on the server. but there are clients which will not send any respone to the server if they accepted or denyed it and there is no timer in vanilla minecraft that will lety say kick the player if he didnt accepted it in less that 10sec. would be cool to have a option to set a timer to kick players who didnt accept or denyd the ressourcepack or fix it in a other way.

**Comment by cittyinthecloud** [comment]
All this does is punish people with good internet, clients would just add the 10 second delay.

**Comment by TetraTheta** [comment, marked as answer]
AFAIK, Purpur freezes player who didn't answered about resource pack download. This would be ideal to hacked client because hacked client can just ignore this kind of prompt and move freely. If server sets god mode to player who didn't answered the prompt, they can play entire game with god mode, which would cause serious problem.

This is what can be happened.

1. Client has good internet speed to download resource pack.
  1. Client accepts the prompt: Everything would be good because player would freeze until he gets the resource pack.
  2. Client denies the prompt: Client would be just kicked. Nothing is wrong about this because it is what that `server.properties` is meant to do.
  3. Client ignores the prompt (with hacked client): Purpur can still freeze player, thus prevents illegal bypass. Client must accept resource pack to play server.
2. Client has poor internet speed to download resource pack.
  1. Accepted: Player will set freeze (god mode) until he gets the resource pack. All he have to do is wait.
  2. Denied: Player will be kicked. Just normal thing.
  3. Ignored (with hacked client): Player will still set freeze because he didn't answered the prompt. Another hacky thing cut offed.

So, I find nothing wrong about this.

**Reply by cittyinthecloud** [reply]
Hacked client who doesn't want the resource pack: Sits there for a few seconds and says "done!"

**Reply by YouHaveTrouble** [reply]
Nothing to be done about this, only wat for server to know that client downloaded resource pack is client response. Also it would break nothing but not load resource pack, which is usually a bad idea for the client to do lol. afaik nothing will break because the process is almost exactly the same as normal resource pack accept.
